drivers/sensor/xbr818/xbr818.c

```diff
@@ -0,0 +1,533 @@
+/*
+ * Copyright (c) 2024 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT phosense_xbr818
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(XBR818, CONFIG_SENSOR_LOG_LEVEL);
+
+#include "xbr818.h"
+#include <zephyr/drivers/sensor/xbr818.h>
+
+static int xbr818_enable_i2c(const struct device *dev)
+{
+	const struct xbr818_config *config = dev->config;
+	int ret;
+
+	if (config->i2c_en.port) {
+		ret = gpio_pin_set_dt(&config->i2c_en, 1);
+		if (ret != 0) {
+			LOG_ERR("%s: could not set i2c_en pin", dev->name);
+		}
+		k_usleep(10);
+		return ret;
+	}
+	return 0;
+}
+
+static int xbr818_disable_i2c(const struct device *dev)
+{
+	const struct xbr818_config *config = dev->config;
+	int ret;
+
+	if (config->i2c_en.port) {
+		ret = gpio_pin_set_dt(&config->i2c_en, 0);
+		if (ret != 0) {
+			LOG_ERR("%s: could not unset i2c_en pin", dev->name);
+		}
+		return ret;
+	}
+	return 0;
+}
+
+static int xbr818_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct xbr818_config *config = dev->config;
+	struct xbr818_data *data = dev->data;
+	int ret;
+
+	if (chan != SENSOR_CHAN_PROX && chan != SENSOR_CHAN_ALL) {
+		LOG_ERR("%s: requesting unsupported channel %i", dev->name, chan);
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_get_dt(&config->io_val);
+	if (ret < 0) {
+		return ret;
+	}
+	data->value = (ret == 1 ? true : false);
```

Comment:
nit, why not just `data->value = rent == 1;`?

Reply:
I do prefer the ternary form.